pkg/registry/apis/playlist/legacy_storage.go

```diff
@@ -62,31 +62,14 @@ func (s *legacyStorage) List(ctx context.Context, options *internalversion.ListO
 		return nil, err
 	}
 
-	limit := 100
-	if options.Limit > 0 {
-		limit = int(options.Limit)
-	}
-	res, err := s.service.Search(ctx, &playlistsvc.GetPlaylistsQuery{
-		OrgId: orgId,
-		Limit: limit,
-	})
+	res, err := s.service.List(ctx, orgId)
 	if err != nil {
 		return nil, err
 	}
 
 	list := &playlist.PlaylistList{}
 	for _, v := range res {
-		p, err := s.service.Get(ctx, &playlistsvc.GetPlaylistByUidQuery{
```

Comment:
^^ this would issue a new query (that makes 2 SQL selects) for each item in the list!